src/pages/api/searchOperators.ts

```diff
@@ -2,17 +2,71 @@ import { NextApiResponse } from 'next';
 import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
 import { SEARCH_OPERATOR_ATTRIBUTE } from '../../constants';
 import { redirectTo, redirectToError } from './apiUtils';
-import { updateSessionAttribute } from '../../utils/sessions';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
 import { removeExcessWhiteSpace } from './apiUtils/validator';
+import { OperatorNameType } from '../../data/auroradb';
 
-// unsure if this interface is needed
-export interface SearchOperatorsWithErrors {
+export interface SearchOperators {
+    selectedOperators: OperatorNameType[];
     errors: ErrorInfo[];
 }
 
 export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
     const errors: ErrorInfo[] = [];
 
+    const { addOperator } = req.body;
+    const refererUrl = req?.headers?.referer;
+    const queryString = refererUrl?.substring(refererUrl?.indexOf('?') + 1);
```

Comment:
Wasnt sure where to put this, but could we add some validation to stop special characters? We're letting the user search with ? and / and . etc currently, would be nice to stop it. I think its doubly important because we track the location of the ? in the query string, so if they search with a ? , it will maybe confuse it.